extension/bearertokenauthextension/bearertokenauth.go

```diff
@@ -177,15 +177,15 @@ func (b *BearerTokenAuth) bearerToken() string {
 // RoundTripper is not implemented by BearerTokenAuth
 func (b *BearerTokenAuth) RoundTripper(base http.RoundTripper) (http.RoundTripper, error) {
 	return &BearerAuthRoundTripper{
-		baseTransport: base,
-		bearerToken:   b.bearerToken(),
+		baseTransport:   base,
+		bearerTokenAuth: b,
```

Comment:
Instead of referencing the whole component, perhaps reference only the bearer token func?

```suggestion
		bearerTokenFunc: b.bearerToken,
```